notifier/tests/test_tasks.py

```diff
@@ -204,3 +206,59 @@ def test_do_forums_digests_user_api_unavailable(self):
                 # should have raised
                 self.fail("task did not give up after exactly 3 attempts")
 
+
+    @override_settings(FORUM_DIGEST_TASK_BATCH_SIZE=10)
+    def test_do_forums_digests_creates_database_entry(self):
+        # patch _time_slice
+        # patch get_digest_subscribers
+        dt1 = datetime.datetime.utcnow()
+        dt2 = dt1 + datetime.timedelta(days=1)
+        with nested(
+            patch('notifier.tasks.get_digest_subscribers', return_value=(usern(n) for n in xrange(11))),
+            patch('notifier.tasks.generate_and_send_digests'),
+            patch('notifier.tasks._time_slice', return_value=(dt1, dt2))
+        ) as (p, t, ts):
+            self.assertEqual(ScheduledForumsDigestTask.objects.count(), 0)
+            task_result = do_forums_digests.delay()
+            self.assertTrue(task_result.successful())
+            self.assertEqual(ScheduledForumsDigestTask.objects.count(), 1)
+            model = ScheduledForumsDigestTask.objects.all()[0]
+            self.assertEqual(model.from_dt, dt1)
+            self.assertEqual(model.to_dt, dt2)
+            self.assertEqual(model.node, platform.node())
+
+
+    @override_settings(FORUM_DIGEST_TASK_BATCH_SIZE=10)
+    def test_do_forums_digests_already_scheduled(self):
+        # patch _time_slice
+        # patch get_digest_subscribers
+        dt1 = datetime.datetime.utcnow()
+        dt2 = dt1 + datetime.timedelta(days=1)
+        dt3 = dt2 + datetime.timedelta(days=1)
+        # Scheduling the task for the first time sends the digests:
+        with nested(
```

Comment:
Just as a side note, `contextlib.nested()` is kind of broken, and Python supports multiple context managers in a single `with` statement natively since version 2.7:

    with patch(…) as p1, patch(…) as p2, patch(…) as p3:
        # some code

You will have to use back slashes for line continuation, since you can't put parens around the list of context managers.